presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoInstaller.java

```diff
@@ -85,21 +94,130 @@ private static void installUsingRpm(String pathToRpm, boolean checkDependencies)
         if (installRpm != 0) {
             throw new PrestoManagerException("Failed to install Presto", installRpm);
         }
-        addTpchCatalog();
+        addTpchCatalog(CATALOG_DIR);
         LOGGER.debug("Successfully installed Presto");
     }
 
-    private static void addTpchCatalog()
+    private static void addTpchCatalog(Path directory)
             throws PrestoManagerException
     {
         try {
-            if (!isDirectory(CATALOG_DIR)) {
-                createDirectory(CATALOG_DIR);
-                write(CATALOG_DIR.resolve("tpch.properties"), "connector.name=tpch".getBytes());
+            if (!isDirectory(directory)) {
+                createDir(directory);
             }
+            write(directory.resolve("tpch.properties"), "connector.name=tpch".getBytes());
         }
         catch (IOException e) {
             throw new PrestoManagerException("Failed to add tpch catalog", e);
         }
     }
+
+    private static void installUsingTar(URL urlToFetchPackage)
+            throws PrestoManagerException
+    {
+        preInstallChecks();
+        File tarTempFile = fetchFileFromUrl(urlToFetchPackage);
+        try {
+            if (isDirectory(TAR_INSTALLATION_DIR)) {
+                deleteDirContents(TAR_INSTALLATION_DIR);
+            }
+            else {
+                createDir(TAR_INSTALLATION_DIR);
+            }
+            untarFile(tarTempFile, TAR_INSTALLATION_DIR);
+        }
+        finally {
+            if (!tarTempFile.delete()) {
+                LOGGER.warn("Failed to delete the temporary file: %s", tarTempFile.toString());
+            }
+        }
+        createDir(TAR_INSTALLATION_DIR.resolve("etc"));
+        addTpchCatalog(TAR_INSTALLATION_DIR.resolve("etc/catalog"));
+        addConfigFiles();
+    }
+
+    private static void preInstallChecks()
+            throws PrestoManagerException
+    {
+        if (Float.valueOf("1.8").compareTo(Float.valueOf(System.getProperty("java.specification.version"))) < 0) {
+            throw new PrestoManagerException("Java version 8 or above is required");
+        }
+    }
+
+    private static void addConfigFiles()
+            throws PrestoManagerException
+    {
+        addConfigProperties();
+        addEnvSh();
+        addJvmConfig();
+        addNodeProperties();
+    }
+
+    private static void addConfigProperties()
+            throws PrestoManagerException
+    {
+        Properties properties = new Properties();
+        properties.setProperty("coordinato", "true");
```

Comment:
Not needed because. I'm just trying to create the files with default values (same as RPM installation). If user wants to modify these `config API` can be used